astropy/units/format/utils.py

```diff
@@ -46,7 +46,7 @@ def get_grouped_by_powers(bases, powers):
     return positive, negative
 
 
-def split_mantissa_exponent(v):
+def split_mantissa_exponent(v, precision=8):
```

Comment:
I think rather than make the format "programmable", one should just allow one to pass it in. I.e., I would suggest making a keyword here `format_spec=".8g"` and use that below. I think the big advantage of that is that it will be much easier to extend (e.g., one can then give `format_spec="+.8g" and get the positive sign, without yet adding another keyword; one could then also adjust `_repr_latex_` later to take into account more of the print options).